coupler/coupler_types.F90

```diff
@@ -1583,7 +2265,8 @@ subroutine CT_redistribute_data_3d(var_in, domain_in, var_out, domain_out, compl
     type(domain2D),           intent(in)    :: domain_out !< The FMS domain for the output structure
     logical,        optional, intent(in)    :: complete   !< If true, complete the updates
 
-    real, pointer, dimension(:,:,:) :: null_ptr3D => NULL()
+    real(r4_kind), pointer, dimension(:,:,:) :: null_ptr3D_r4 => NULL()
+    real(r8_kind), pointer, dimension(:,:,:) :: null_ptr3D_r8 => NULL()
```

Comment:
#1369

Reply:
The way these are being used here might be better to leave them as a pointer.  Otherwise, you'd need to have them be a (1,1,1) array in order to satisfy the interface rules for mpp_redistribute.